ui/app/src/Views/AlertList/Updates.elm

```diff
@@ -26,20 +29,44 @@ update msg ({ groupBar, alerts, filterBar, receiverBar, alertGroups } as model)
             )
 
         AlertsFetched listOfAlerts ->
-            ( { model
-                | alerts = listOfAlerts
-                , groupBar =
+            let
+                ( groups_, groupBar_ ) =
                     case listOfAlerts of
                         Success ungroupedAlerts ->
-                            { groupBar
-                                | list =
-                                    List.concatMap (.labels >> Dict.toList) ungroupedAlerts
-                                        |> List.map Tuple.first
-                                        |> Set.fromList
-                            }
-
-                        _ ->
-                            groupBar
+                            let
+                                groups =
+                                    ungroupedAlerts
+                                        |> Utils.List.groupBy
+                                            (.labels >> Dict.toList >> List.filter (\( key, _ ) -> List.member key groupBar.fields))
+                                        |> Dict.toList
+                                        |> List.map
+                                            (\( labels, alerts_ ) ->
+                                                AlertGroup (Dict.fromList labels) { name = "unknown" } alerts_
+                                            )
+
+                                newGroupBar =
+                                    { groupBar
+                                        | list =
+                                            List.concatMap (.labels >> Dict.toList) ungroupedAlerts
+                                                |> List.map Tuple.first
+                                                |> Set.fromList
+                                    }
+                            in
+                            ( Success groups, newGroupBar )
+
+                        Initial ->
+                            ( Initial, groupBar )
+
+                        Loading ->
+                            ( Loading, groupBar )
+
+                        Failure e ->
+                            ( Failure e, groupBar )
```

Comment:
This is sorting the alerts into alert groups immediately when they're returned from the API. I wasn't sure how to have a pattern match `Initial`, `Loading`, and `Failure e`, because they're technically of type `ApiData GettableAlerts`. If there's a way to compact this, that would be awesome.

Reply:
I think this is fine.